### PR TITLE
Enabling tests which would have prevented a bug

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 1]
```diff
@@ -168,6 +180,12 @@ public EndToEndTests(ServerFixture<Startup> serverFixture, ITestOutputHelper out
                     logger.LogInformation("Disposed Connection");
                 }
             }
+
+            bool base64Encoded(TransferMode transferMode, IConnection connection)
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: nit: casing this isn't typescript 😉

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: LOL

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Fixed.

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 2]
```diff
@@ -301,5 +319,14 @@ public static IEnumerable<object[]> MessageSizesData
                 new object[] { TransportType.ServerSentEvents },
                 new object[] { TransportType.LongPolling }
             };
+
+        public static IEnumerable<object[]> TransportTypesXTransferModes()
```
Comment (muratg) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: XTransfer?

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: read: TransportTypes times TransferModes

Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Maybe call it And

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: The thing is that this is not *And* - this is Cartesian product. We also have [one more instance](https://github.com/aspnet/SignalR/blob/3a1d4c5dd66cdcccbc61dc8e530784c8b79b9b1b/test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs#L220) of this so if we want to change the convention it should be in both places.

Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: I know but I don't think it matters.

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: ok - I can change isn't that important..

Comment (muratg) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Thanks for the Xplanation :)

File: src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs [hunk 3]
```diff
@@ -120,6 +120,14 @@ public class HubEndPoint<THub> : IInvocationBinder where THub : Hub
                             ? (IDataEncoder)Base64Encoder
                             : PassThroughEncoder;
 
+                        var transferMode = connection.Features.Get<ITransferModeFeature>() ??
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs]: nit: rename to transferModeFeature

File: src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs [hunk 4]
```diff
@@ -120,6 +120,14 @@ public class HubEndPoint<THub> : IInvocationBinder where THub : Hub
...
+                            throw new InvalidOperationException("Unable to read transfer mode.");
+
+                        transferMode.TransferMode =
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs]: Where is this used? Or are we just updating it to be correct? I can't see who uses this thing.

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs]: https://github.com/aspnet/SignalR/blob/29f9c54b864d3f86522b15651acf3b6adc57c8be/src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs#L181-L183
- I must have lost it in one of rebases/merges.

**Comment by moozzyk** [comment]
Addressed feedback

**Pull request merged**